contrib/buildrpm/README

```diff
@@ -0,0 +1,90 @@
+
+# Copyright (c) 2016      Cisco Systems, Inc.  All rights reserved.
+#
+# Written by: Landon Curt Noll, Martin Kontsek
+#
+# Permission to use, copy, modify, and distribute this software and
+# its documentation for any purpose and without fee is hereby granted,
+# provided that the above copyright, this permission notice and text
+# this comment, and the disclaimer below appear in all of the following:
+#
+#       supporting documentation
+#       source copies
+#       source works derived from this source
+#       binaries derived from this source or from derived source
+#
+# Landon Curt Noll, Martin Kontsek DISCLAIMS ALL WARRANTIES WITH REGARD
+# TO THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+# AND FITNESS. IN NO EVENT SHALL Landon Curt Noll AND Martin Kontsek BE
+# LIABLE FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+
+=================================================================
+
+Script "buildrpmLibfabric.sh" can be used to build RPM from the
+stable Libfabric source tarball. Script was tested on RedHat 7.
+
+You need to supply one mandatory parameter to the script:
+filename of the Libfabric tarball.
+There are several voluntary parameters available:
+
+Provider parameters:
+-a
+    Build with psm provider support.
+-b
+    Build with psm2 provider support.
+-c
+    Build with sockets provider support.
+-e
+    Build with verbs provider support.
+-f
+    Build with usnic provider support.
+-g
+    Build with mxm provider support.
+-i
+    Build with gni provider support.
+-j
+    Build with xpmem (gni provider) support.
+-k
+    Build with udp provider support.
+-l
+    Build with rxm provider support.
+-p
+    Build with rxd provider support.
```

Comment:
You mean to remove all the options for providers and just add user option like this:

> -p <provider_name>
>    Build with <provider_name> provider support.

User would be able to specify this option multiple times to enable multiple providers. 
There could be also similar option to disable provider.